terraform/environments/mlra/modules/ALB/main.tf

```diff
@@ -0,0 +1,89 @@
+module "lb-access-logs-enabled" {
+  source = "github.com/ministryofjustice/modernisation-platform-terraform-loadbalancer"
+  providers = {
+    aws.bucket-replication = aws.bucket-replication
+  }
+
+  vpc_all                    = var.vpc_all
+  application_name           = var.application_name
+  public_subnets             = var.public_subnets
+  region                     = var.region
+  enable_deletion_protection = var.enable_deletion_protection
+  idle_timeout               = var.idle_timeout
+  force_destroy_bucket       = var.force_destroy_bucket
+  tags                       = var.tags
+  account_number             = var.account_number
+  loadbalancer_ingress_rules = local.loadbalancer_ingress_rules
+  loadbalancer_egress_rules  = local.loadbalancer_egress_rules
+}
+
+locals {
+  loadbalancer_ingress_rules = {
+    "cluster_ec2_lb_ingress" = {
+      description     = "Cluster EC2 loadbalancer ingress rule"
+      from_port       = 443
+      to_port         = 443
+      protocol        = "tcp"
+      cidr_blocks     = [var.ingress_cidr_block]
+      security_groups = []
+    }
+  }
+  loadbalancer_egress_rules = {
+    "cluster_ec2_lb_egress" = {
+      description     = "Cluster EC2 loadbalancer egress rule"
+      from_port       = 0
+      to_port         = 0
+      protocol        = "-1"
+      cidr_blocks     = ["0.0.0.0/0"]
+      security_groups = []
+    }
+  }
+}
+
+resource "aws_lb_listener" "alb_listener" {
+  load_balancer_arn = module.lb-access-logs-enabled.load_balancer.arn
+  port              = var.listener_port
+  protocol          = "HTTP"
```

Comment:
Would it be worth moving this hard-coded value out to the module call?

Reply:
var.listener_protocol has been added to replace the hardcoded value